gix/tests/gix/revision/spec/from_bytes/regex.rs

```diff
@@ -95,20 +95,38 @@ mod find_youngest_matching_commit {
     fn regex_matches() {
         let repo = repo("complex_graph").unwrap();
 
-        assert_eq!(
-            parse_spec(":/mes.age", &repo).unwrap(),
-            Spec::from_id(hex_to_id("ef80b4b77b167f326351c93284dc0eb00dd54ff4").attach(&repo))
-        );
+        // On the CI linux workflow, archived baseline aren't used - instead fixtures will be re-evaluated.
+        // As of Git 2.47, its behaviour changed which makes the following assertion fail.
+        // We decided to just ignore it until it's clear that this isn't a bug - obviously the traversal order changed.
+        let is_in_test_ci_workflow = is_ci::cached() && cfg!(target_os = "linux");
```

Comment:
If this were to also require that `GIX_TEST_IGNORE_ARCHIVES` be set (or, if that is judged too complex, then to replace the `linux` check with such a check), then `parse_spec` would continue to be used in the CI `test_fast` jobs that don't need it because their baselines were computed with an earlier versions of Git and agree with the current gitoxide behavior, while `parse_spec_no_baseline` would be used in the CI `test` job that needs it.

Reply:
Thank you, that's a much better selector actually.
This PR is a good example for quick & sloppy, but thanks to you it's not going to be merged that  way :).